hedgedocs/docs/hedgehog-engine/sonic2010/files/setobject.md

```diff
@@ -0,0 +1,95 @@
+---
+
+# Set Object
+
+The Set Object file format is used in Sonic Colors in order to define stage object layout, serving the same purpose as "Set" and "setdata" XML files in Sonic Unleashed and Sonic Generations respectively. However, unlike in those games, this data is actually stored in binary format for greater efficiency, with the same practice being carried forward in all future titles by Sonic Team.
+
+## Set Object Format
+
+Set object data is stored in "stgxxx_obj_xx.orc" BINA containers, which can be found in "set" folder in "sonic2010_0.cpk". The objects themselves are arranged into 8 "layers", separated into files with suffixes from 00 to 07 in the "set" folder for each stage. In the original Wii release, the data is stored in big-endian format.
+
+```csharp
+class SetObjectData {
+    uint signature = 1397703242; // "SOBJ" ASCII in big-endian.
+    uint version = 1; // Always "1".
+    uint objectDictCount; // The number of unique object types.
+    uint objectDictOffset; // The non-absolute offset to the object dictionary array.
+    uint bvhTreeOffset; // The non-absolute offset to the bounding volume hierarchy tree. This is unused in Colors.
+    uint objectOffsetArrayOffset; // The non-absolute offset to the array of offsets to all the objects stored in the file.
+    uint objectCount; // The number of objects stored in the file.
+    uint bvhTreeNodeCount; // The number of nodes in the bounding volume hierarchy tree.
+    uint objectInstanceCount; // The total number of object instances in the object instance array.
+
+    SetObjectDictNode[objectDictCount] objectDictArray; // The dictionary of object types;
+    SetObjectIndicesArrayNode[objectDictCount] objectIndicesArray; // An array of nodes, that each store indices to objects in the object array, grouped by object type.
+    uint[objectCount] objectOffsetArray; // An array of offsets to all of the objects stored in the file, indexed into by SetObjectIndicesArrayNode entries.
+    SetObject[objectCount] objectArray; // The "array" of objects. The term "array" is used loosely here as the objects are not fixed size, due to the custom parameters that can be defined for them.
+    SetObjectInstance[objectInstanceCount] objectInstanceArray; // The array of object instances.
+}
+
+// One node for every object type in the SOBJ file.
+class SetObjectDictNode {
+    uint objectTypeNameOffset; // The non-absolute offset to the object type name in the BINA string table.
+    uint objectIndicesCount; // The number of indices in the object indices array that are of the corresponding type.
+    uint objectIndicesOffset; // The non-absolute offset to the array of object indices for the corresponding type.
+}
+
+// One node for every object type in the SOBJ file.
+class SetObjectIndicesArrayNode {
+    uint[objectTypeCount] objectIndicies; // An array of indices of all the objects of the corresponding type in the object indices array.
```

Comment:
I think you meant to refer to `objectIndicesCount` instead of `obejctTypeCount`, right?